engine/schema/src/com/cloud/vm/dao/VMInstanceDao.java

```diff
@@ -150,4 +150,9 @@
     VMInstanceVO findVMByHostNameInZone(String hostName, long zoneId);
 
     boolean isPowerStateUpToDate(long instanceId);
+
+    /**
+     * List running VMs which host and lastHost id are the same
+     */
+    List<VMInstanceVO> listRunningVmsByHostAndLastHostSameId(long hostId);
```

Comment:
How about renaming the method to listRunningVmsByHostEqualsLastHost(), so we don't need the javadoc anymore?